### PR TITLE
Fix/upgraded packages and fixed multipart

### PR DESCRIPTION
**Pull request by andylockran**
As requested in #33 , I have rebased and set two tidy commits to upgrade the security errata and enable MultiPart options on uploads.

This fixes the issue #32.

According to the documentation, only the ETag and PartNumber should be part of the parts array.

https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#completeMultipartUpload-property

To show this breaking, add 'ServerSideEncryption: "AES256" to your initial options - and despite it being a valid parameter, you'll see the code bum out at this point.

var S3 = require('aws-sdk').S3,
    S3S = require('s3-streams');

var bucket="test_bucket_for_bug_32";

var params = {
    Bucket: bucket,
    Key: "test_file.txt",
    ServerSideEncryption: "AES256"
}

var s = S3S.WriteStream(s3client, params);

s.write("Example test content.. Should probably also test with longer content..")
s.end();

### CODE REVIEW

**Comment by andylockran** [comment]
Aligning with upstream

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 Support for streaming reads and writes from and to S3 using Amazon's native API.
 
-![build status](http://img.shields.io/travis/izaakschroeder/s3-streams/master.svg?style=flat)
+![build status](http://img.shields.io/travis/andylockran/s3-streams/master.svg?style=flat)
```
Comment (izaakschroeder) [inline README.md]: Is this intentional? 😄

**Review by izaakschroeder — COMMENTED** [review]
Only one minor thing, otherwise looks good to me!

**Comment by andylockran** [comment]
Nope. Will do and amend the commit once I'm home. 60 mins.

File: README.md (hunk 1 above)
Comment (andylockran) [reply README.md]: Nope - fixed in latest.

**Comment by andylockran** [comment]
Ok, that should be it all pukka!

**Comment by izaakschroeder** [comment]
Still seeing that one line there @andylockran 😄

**Comment by andylockran** [comment]
I understand why you don't accept contributions... 🥇 

Finally got it fixed.  You going to up the version number for npm?

**Comment by izaakschroeder** [comment]
Ha, looks good thanks! 😄 I'll bump it up 🎉

**Comment by izaakschroeder** [comment]
All set in `0.4.0`. 🚀

**Pull request merged**